src/main/java/com/thealgorithms/sorts/TopologicalSort.java

```diff
@@ -0,0 +1,170 @@
+package com.thealgorithms.sorts;
+
+import java.util.*;
+
+/**
+ * The Topological Sorting algorithm linearly orders a DAG or Directed Acyclic Graph into
+ * a linked list. A Directed Graph is proven to be acyclic when a DFS or Depth First Search is
+ * performed, yielding no back-edges.
+ *
+ * https://en.wikipedia.org/wiki/Topological_sorting
+ *
+ * @author Jonathan Taylor (https://github.com/Jtmonument)
+ * Based on Introduction to Algorithms 3rd Edition
+ */
+public class TopologicalSort {
+
+     /*
+     * Enum to represent the colors for the depth first search
+     * */
+    private enum Color {
+        WHITE, GRAY, BLACK
+    }
+
+     /*
+     * Class to represent vertices
+     * */
+    private static class Vertex {
+        /*
+        * Name of vertex
+        * */
+        public final String label;
+
+        /*
+        * Weight of vertex
+        * (more accurately defined as the time that a vertex has begun a visit in DFS)
+        * */
+        public int weight;
+
+        /*
+        * The time that the vertex has finished a visit in DFS
+        * */
+        public int finished;
+
+        /*
+        * π parent of the vertex
+        * */
+        public Vertex predecessor;
+
+        /*
+        * Represents the category of visit in DFS
+        * */
+        public Color color = Color.WHITE;
+
+        /*
+        * The array of names of descendant vertices
+        * */
+        public final ArrayList<String> next = new ArrayList<>();
+
+        public Vertex(String label) {
+            this.label = label;
+        }
+     }
+
+     /*
+     * Graph class uses the adjacency list representation
+     * */
+    static class Graph {
+
+        /*
+         * Adjacency list representation
+         * */
+        private final HashMap<String, Vertex> adj = new LinkedHashMap<>();
+
+        /*
+         * Function to add an edge to the graph
+         * */
+        public void addEdge(String label, String... next) {
+            adj.put(label, new Vertex(label));
+            if (!next[0].isEmpty())
+                Collections.addAll(adj.get(label).next, next);
+        }
+    }
+
+    static class BackEdgeException extends RuntimeException {
+
+        public BackEdgeException(String backEdge) {
+            super("This graph contains a cycle. No linear ordering is possible. " + backEdge);
+        }
+
+    }
+
+    /*
+     * The linked list that will contain the vertices in linear order.
+     * */
+    private final LinkedList<String> list = new LinkedList<>();
```

Comment:
Please just create a new list in the sort function and return it from it instead of storing the list in a field